src/lab14.py

```diff
@@ -1214,28 +1217,27 @@ def render(self):
 
         self.measure_render.stop()
 
-    def apply_focus(self, node):
+    def focus_element(self, node):
         if self.focus:
             self.focus.is_focused = False
         self.focus = node
         if node:
-            if node.tag == "input":
-                node.attributes["value"] = ""
             node.is_focused = True
         self.set_needs_render()
 
     def activate_element(self, elt):
-        if elt.tag == "a" and "href" in elt.attributes:
+        if elt.tag == "input":
```

Comment:
This is called by `enter()` also; is this behavior appropriate in that situation?

Reply:
Well, I do call it out in the text as a bit of a quirk. In a normal browser `enter` in a text area submits the form, but our browser has never had that behavior. And we need the clearing behavior. It's definitely better to have it in activate than in focus; otherwise, just tabbing through a form requires re-entering every text entry you go through.